--- tsidump.c	2004-05-12 23:30:19.000000000 +0800
+++ tsidump-chewing.c	2004-05-12 23:29:12.000000000 +0800
@@ -65,32 +65,23 @@
     }
     i++;
     len = strlen((char *)tsi->tsi)/2;
-    fprintf(fp, "%s", tsi->tsi);
-    if (ref) {
-      fprintf(fp, " %ld", tsi->refcount);
-    }
     if (tsiyin) {
       ZuYinSymbolSequence zs = NULL;
-      int begin = 0;
 
-      fprintf(fp, " ");
       for (j = 0; j < tsi->yinnum; j++) {
+	if (len > 7)
+	  continue;
+	fprintf(fp, "%s %ld\t", tsi->tsi, tsi->refcount);
         for (k = 0; k < len; k++) {
           zs = tabeYinToZuYinSymbolSequence(tsi->yindata[j*len+k]);
           if (zs) {
-            if (begin) {
-              fprintf(fp, "¡@");
-            }
-            else {
-              begin = 1;
-            }
-            fprintf(fp, "%s", zs);
+	    fprintf(fp, "%s ", zs);
             free(zs);
           }
         }
+	fprintf(fp, "\n");
       }
     }
-    fprintf(fp, "\n");
   }
 
   db->Close(db);
@@ -109,8 +100,8 @@
   char *db_name, *op_name;
 
   db_name = op_name = (char *)NULL;
-  ref = 0;
-  tsiyin = 0;
+  ref = 1;
+  tsiyin = 1;
 
   while ((ch = getopt(argc, argv, "d:f:ry")) != -1) {
     switch(ch) {
